test/Microsoft.AspNet.TestHost.Tests/ClientHandlerTests.cs

```diff
@@ -234,5 +243,33 @@ public Task ExceptionBeforeFirstWriteIsReported()
             var ex = await Assert.ThrowsAsync<HttpRequestException>(() => response.Content.ReadAsStringAsync());
             Assert.IsType<InvalidOperationException>(ex.GetBaseException());
         }
+
+        private class DummyApplication : IHttpApplication<Context>
+        {
+            Func<Context, Task> _application;
+
+            public DummyApplication(Func<Context, Task> application)
```

Comment:
Yes, DummyApplication needs to be `IHttpApplication<Context>`, but the constructor can take `Func<HttpContext, Task>` to simplify the tests. The only difference is that `return _application(context);` changes to `return _application(context.HttpContext);`